doc/metrics.md

```diff
@@ -0,0 +1,80 @@
+# Metrics
+
+etcd-backup-restore uses [Prometheus][prometheus] for metrics reporting. The metrics can be used for real-time monitoring and debugging. It won't persist its metrics; if a member restarts, the metrics will be reset.
+
+The simplest way to see the available metrics is to cURL the metrics endpoint `/metrics`. The format is described [here](http://prometheus.io/docs/instrumenting/exposition_formats/).
+
+Follow the [Prometheus getting started doc][prometheus-getting-started] to spin up a Prometheus server to collect etcd metrics.
+
+The naming of metrics follows the suggested [Prometheus best practices][prometheus-naming]. All etcd-backup-restore related metrics are put under namespace `etcdbr`.
+
+### ETCD metrics
+
+The metrics under the `etcd` prefix/namespace are carried forward from etcd library that we use. These metrics does not include details of the `etcd` deployment on which `etcd-backup-restore` utility works. Instead, it helps the monitoring the `embedded etcd` we spawn while restoration process.
+
+
+### Snapshot
+
+These metrics describe the status of the snapshotter. In order to detect outages or problems for troubleshooting, these metrics should be closely monitored. The below mentioned metrics are listed as collection of series using prometheus labels `kind` and `succeeded`. `Kind` label indicates the snapshot kind i.e. full snapshot or incremental/delta snapshot in the context. And succeeded indicates whether the metrics is for successful operation or erroneous operation.
+
+| Name | Description | Type |
+|------|-------------|------|
+| etcdbr_snapshot_duration_seconds | Total latency distribution of saving snapshot to object store. | Histogram |
+| etcdbr_snapshot_gc_total | Total number of garbage collected snapshots. | Counter |
+| etcdbr_snapshot_latest_revision | Revision number of latest snapshot taken. | Gauge |
```

Comment:
Why can't `etcdbr_snapshot_latest_revision` and `etcdbr_snapshot_latest_timestamp` be of metric type `counter` instead of `gauge`? Because these values always increase right? In what case(s) would they decrease?

Reply:
It could be possible with counter. But revision values is grabbed from etcd and etcd-backup-restore is not maintainer of it. From etcd-backup-restore's point of view it always value update rather some know increment in revision number. Another reason, I went for `Guage` over `Counter`, is, looking at the API counter need the delta from previous update, which looks little complex when implemented for etcd revision metrics scenario.